caddy: 2.7.6 -> 2.8.4

Nanotwerp opened the pull request:
## Description of changes
Updates `caddy` from version [`2.7.6`](https://github.com/caddyserver/caddy/releases/tag/v2.7.6) to version [`2.8.4`](https://github.com/caddyserver/caddy/releases/tag/v2.8.4). This commit also adds the `nobadger` build tag, which prevents the `badger-db` implementation of `NoSQL` from being compiled. This reduces the binary size of `caddy`, and is being used by `caddy`'s upstream ever since `2.8.0`.
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4010

flokli reviewed (dismissed):
I'm not sure we should actively migrate code over to another code formatter. At least, I'd leave it up to the individual package maintainers for now.

Squashing it into the same commit that does a bump makes this very hard to review. Please make it two separate commits at least, or leave the reformatting up to the maintainers, or a tree wide formatting, if that happens at some point.

Nanotwerp committed:
caddy: 2.7.6 -> 2.8.4

Starting with v2.8.0, Caddy recommends building with the `nobadger`
build tag.

The build tag makes go skip the compilation of the badger-db
implementation from smallstep/nosql, as it isn't used by Caddy and thus
slightly shrinks the resulting binary size.

https://github.com/caddyserver/caddy/releases/tag/v2.8.4

https://github.com/caddyserver/caddy/releases/tag/v2.8.3

https://github.com/caddyserver/caddy/releases/tag/v2.8.2

https://github.com/caddyserver/caddy/releases/tag/v2.8.1

https://github.com/caddyserver/caddy/releases/tag/v2.8.0

diff: https://github.com/caddyserver/caddy/compare/v2.7.6...v2.8.4

Nanotwerp commented:
> I'm not sure we should actively migrate code over to another code formatter. At least, I'd leave it up to the individual package maintainers for now.
> 
> Squashing it into the same commit that does a bump makes this very hard to review. Please make it two separate commits at least, or leave the reformatting up to the maintainers, or a tree wide formatting, if that happens at some point.

Done! I made them separate commits.

flokli commented:
Thanks! Bump changes LGTM, but let's have the package maintainers decide about the PR.

emilylange commented on pkgs/by-name/ca/caddy/package.nix:
```diff
@@ -24,39 +25,42 @@ buildGoModule {
+    "-w"
     "-X github.com/caddyserver/caddy/v2.CustomVersion=${version}"
   ];
 
```
Please add the `nobadger` build tag, as mentioned in the release notes and https://github.com/caddyserver/caddy/pull/6031.

Also see https://github.com/caddyserver/xcaddy/commit/d6f5a3e6c3dd820060bd2ec75c8fa67a6cfddb45 and https://github.com/caddyserver/dist/commit/03b7609b4bd9514c0bd84e9b903e9b8573651d78.

```suggestion

  tags = [
    "nobadger"
  ];

```

Nanotwerp replied:
Done, thank you! I undid the `nixfmt-rfc-style` conversion and added the `nobadger` tag.

emilylange commented on pkgs/by-name/ca/caddy/package.nix:
I would prefer it if you could drop the `nixfmt-rfc-style` altogether.
It adds a lot of noise, the formatter is technically still not stable.
Please just wait for the treewide change.

emilylange reviewed (changes requested):
Oh and while you are at it, could you also please add at the very least two links to each of the version's release notes as hinted in our contributing guidelines?

https://github.com/NixOS/nixpkgs/blob/342ac0a9895ef24ab2725d829da96670abae2f09/pkgs/README.md#commit-conventions

---

https://github.com/caddyserver/caddy/releases/tag/v2.8.0

https://github.com/caddyserver/caddy/releases/tag/v2.8.1

emilylange commented:
CI check fails because of a trailing comma and still missing the release note links in the commit message.

It's fine if you only change the CI check failure. I can squash-merge your commit and fix the commit message that way.

Nanotwerp commented:
> CI check fails because of a trailing comma and still missing the release note links in the commit message.
> 
> It's fine if you only change the CI check failure. I can squash-merge your commit and fix the commit message that way.

All done! Sorry for the formatting issues and thousands of force-pushes!

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1707

Nanotwerp commented:
Bumped up to `2.8.2`.

emilylange commented:
You previously had a `vendorHash` mismatch, but you seems to have fixed it as part of the `2.8.2`. That's good.
Now it actually builds :)

However, `2.8.3` will get released soon, so I'd propose to wait for it first.

We could merge `2.8.2` but that would leave us with an unnecessarily buggy version 🫣

> and thousands of force-pushes!

No worries, even thousands of force-pushes are fine.
Arguably GitHub does not handle it all too well, but nixpkgs decided to use force-pushes anyway, so that just how it is :)

fin444 commented:
https://github.com/caddyserver/caddy/releases/tag/v2.8.4

Nanotwerp commented:
Updated to `2.8.4`!

emilylange commented on pkgs/by-name/ca/caddy/package.nix:
```diff
@@ -57,7 +57,7 @@ buildGoModule {
       --zsh <($out/bin/caddy completion zsh)
   '';
-
+  tags = [ "nobadger" ];
```
Is there a particular reason why you added `tags` all the way down at line 60, between `postInstall` and `passthru.tests` instead of somewhere at the top, best probably right below `ldflags` but before `nativeBuildInputs`?

If there isn't a reason, please move it up.

It's really confusing and easy to overlook otherwise.

If you feel fancy, you can add a comment why `nobadger` is needed:


```suggestion
  # matches upstream
  tags = [ "nobadger" ];
```

Nanotwerp replied:
Done, thank you!

emilylange reviewed (changes requested):
How about the following commit message instead:

```
caddy: 2.7.6 -> 2.8.4

Starting with v2.8.0, Caddy recommends building with the `nobadger`
build tag.

The build tag makes go skip the compilation of the badger-db
implementation from smallstep/nosql, as it isn't used by Caddy and thus
slightly shrinks the resulting binary size.

https://github.com/caddyserver/caddy/releases/tag/v2.8.4

https://github.com/caddyserver/caddy/releases/tag/v2.8.3

https://github.com/caddyserver/caddy/releases/tag/v2.8.2

https://github.com/caddyserver/caddy/releases/tag/v2.8.1

https://github.com/caddyserver/caddy/releases/tag/v2.8.0

diff: https://github.com/caddyserver/caddy/compare/v2.7.6...v2.8.4
```

flokli commented:
Please check the failing CI output, you now introduced trailing whitespace.

Nanotwerp commented:
> Please check the failing CI output, you now introduced trailing whitespace.

Fixed. The text editor I use needs to get `EditorConfig` support QUICKLY.
